src/util/check-supported-platforms.js

```diff
@@ -0,0 +1,7 @@
+export function checkSupportedPlatforms(pkg) {
+  const osList = require(`${pkg}/package.json`).os || [];
```

Comment:
This means that makers with packages that don't have an `os` entry will only be available when building for the host platform. This is not what we want for `zip` and `squirrel` (and maybe others?), which should work on any platform.

Maybe the best would be to make this ``` || [ 'win32', 'darwin', 'linux', 'mas' ]``` (or put that array in some module we can include) so packages without an `os` list are available everywhere (because they presumably don't have any platform dependencies)? Then packages that probably should have an `os` entry but don't (like `electron-installer-dmg`) would just have to be special cased in their maker to not use this method, but to hard-code the platforms that we know they actually support.